src/dispatch/auth/permissions.py

```diff
@@ -291,3 +293,71 @@ def has_required_permissions(
         if current_incident.commander:
             if current_incident.commander.individual.email == current_user.email:
                 return True
+
+
+# Cases
+
+
+class CaseViewPermission(BasePermission):
+    def has_required_permissions(
+        self,
+        request: Request,
+    ) -> bool:
+        current_case = case_service.get(
+            db_session=request.state.db, case_id=request.path_params["case_id"]
+        )
+
+        if not current_case:
+            return False
+
+        if current_case.visibility == Visibility.restricted:
+            return any_permission(
+                permissions=[
+                    OrganizationAdminPermission,
+                    CaseAssigneePermission,
+                    CaseReporterPermission,
+                ],
+                request=request,
+            )
+        return True
+
+
+class CaseAssigneePermission(BasePermission):
+    def has_required_permissions(
+        self,
+        request: Request,
+    ) -> bool:
+        current_user = get_current_user(request=request)
+        current_case: Case = case_service.get(
+            db_session=request.state.db, incident_id=request.path_params["case_id"]
+        )
+        if not current_case:
+            return False
+
+        if current_case.assignee:
+            if current_case.assignee.individual.email != current_user.email:
```

Comment:
These could be written more clearly but defining a `has_permission = False` at the beginning and then only having conditionals that would change modify `has_permission = True` and returning the value of `has_permission`.

Reply:
Much cleaner, thanks!